buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -107,6 +107,8 @@
         "is a pre-release version of Elasticsearch",
         "max virtual memory areas vm.max_map_count"
     );
+    private static final String LINUX_DARWIN_HOSTNAME_OVERRIDE = "LinuxDarwinHostname";
```

Comment:
Is there any value  in providing different values per platform here ? Could we provide one and avoid platform dependent checks in the test ?

Reply:
Mainly I wanted to convince myself that I understood how the batch scripts set the hostname variable. For Windows, you might think that setting a `$HOSTNAME` environment variable would do the trick, since that's what the bootstrap code ultimately reads, but that would be wrong, since `elasticsearch.bat` will overwrite `$HOSTNAME` with the value in `$COMPUTERNAME` before starting the application.

Since the different platforms determine the host name differently, I think there's value in demonstrating that in an integration test. On the other hand, I see the value in platform independence for these tests (as opposed to the specifically platform-dependent OS tests). Let me know what you think. I will be happy to change this if you'd like me to (and @rjernst, our other reviewer, agrees).

Reply:
That's a good point. We also test that HOSTNAME doesn't influence Windows this way. 
I would frop "LIUX_DARWIN" and "WINDOWS" from the constant names since we always set them.